src/remind-user.ts

```diff
@@ -90,56 +152,41 @@ const checkAndRemindUsers = async (
   const userList = await db.getActiveUserList({
     activeSince: dateFns.subDays(instant, DAYS_SINCE_LAST_POST_CUT_OFF),
   })
+
   if (userList instanceof Error) {
     return userList
   }
 
   const result = await errorListBoundary(async () =>
     Promise.all(
       userList.map(async (user): Promise<void | Error> => {
-        const userTime = formatDateAsTime({
+        const userDate = formatDateAsISODate({
           instant,
           timeZone: user.timeZone,
         })
-        const userDate = formatDateAsISODate({
+        const userTime = formatDateAsTime({
           instant,
           timeZone: user.timeZone,
         })
-        const isWeekend = dateFns.isWeekend(dateFns.parseISO(userDate))
 
-        const dailyReminderTime =
-          user.dailyReminderTime ?? HANDOVER_DAILY_REMINDER_TIME
+        const reminderNeededToday = await isReminderNeededToday({
+          user,
+          userDate,
+          userTime,
+        })
 
-        if (userTime >= dailyReminderTime && !isWeekend) {
-          const post = await db.getPostWithItems({
+        if (reminderNeededToday) {
+          await remindUsers({ web, user, userDate })
+        } else {
+          await addPostItem({
+            web,
             userId: user.id,
-            date: userDate,
+            postDate: userDate,
+            postTitle: user.name,
+            channel: HANDOVER_CHANNEL,
+            ts: userTime,
+            text: `🫥 ${user.name} is not working today 🫥`,
```

Comment:
I've tested `reminderNeededToday`, post creation, and messaging to the handover channel. I couldn't test E2E scenarios though... I guess it should be alright since this is an internal too, so deploying and seeing how it goes shouldn't be risky. What do you reckon?

Reply:
I think we shouldn't do the automated "user is not working today" in this PR -- it's a bit more complicated than it appears.

The `checkAndRemindUsers` function is called every minute -- so currently this code posts the same message over and over again...

![image](https://github.com/Runn-Fast/handover/assets/1140167/808e8ef1-c5dc-4610-a92b-40dbc695880e)